src/algebra/pointwise.lean

```diff
@@ -576,7 +576,7 @@ section monoid
 
 /-- An alias for `set α`, which has a semiring structure given by `∪` as "addition" and pointwise
   multiplication `*` as "multiplication". -/
-@[derive inhabited] def set_semiring (α : Type*) : Type* := set α
+@[derive [inhabited, partial_order, order_bot]] def set_semiring (α : Type*) : Type* := set α
 
 /-- The identitiy function `set α → set_semiring α`. -/
 protected def up (s : set α) : set_semiring α := s
```

Comment:
The names are *really* mystifying. `s.up` could mean a zillion other things. Also, you don't want such a definition, but rather `set α ≃ set_semiring α` (which can be used the same way except that you can now use dot notation for goodies), maybe decorated one way or another. Could this be fixed in a followup PR?

Reply:
This is definitely not in scope for this PR though

Reply:
Yup, just thought I should mention it somewhere.

Reply:
I have a follow up PR where I add some further instances on `set_semiring`.

Yael, I do not understand your comment: do you want to prove that the "identity" is an isomorphism of as many structures as we can think of?  If so, I may give it a try in a third PR.

Reply:
Or, at the very least, I can add a TODO in the open PR.